### PR TITLE
Start using individual server instances in tests

### PR DESCRIPTION
**Pull request by mpfeil**
#### Motivation
I just started using `got` and really enjoy it`s simplicity. So I would love to contribute and picked up #732.

#### Changes
Started using `create-test-server` like suggested and started rewriting tests for `http.js`.
Maybe you can check the changes and provide some feedback if it fits what you had on mind.

#### Checklist

- [x] I have read the documentation.
- [x] I have included a pull request description of my changes.
- [ ] I have included some tests.
- [ ] If it's a new feature, I have included documentation updates.

### CODE REVIEW

File: test/helpers/with-server.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+'use strict';
+const createTestServer = require('create-test-server');
+
+exports.host = 'localhost';
+const {host} = exports;
```
Comment (szmarczak) [inline test/helpers/with-server.js]: This is not needed. You can simply do `s.host = 'localhost';`

**Comment by szmarczak** [comment]
If you want to convert the other tests, feel free. Or, if you're currently busy I'll try to do it, just let me know :)

File: test/helpers/with-server.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+'use strict';
+const {URL} = require('url');
+const createTestServer = require('create-test-server');
+
+exports.withServer = async (t, run) => {
+	const s = await createTestServer();
+
+	s.host = (new URL(s.url)).hostname;
```
Comment (szmarczak) [inline test/helpers/with-server.js]: @sindresorhus Shouldn't it be named `s.hostname`?

**Comment by mpfeil** [comment]
@szmarczak I can work on some more tests in the next days

File: test/helpers/with-server.js (hunk 2 above)
Comment (sindresorhus) [reply test/helpers/with-server.js]: Yes

**Comment by sindresorhus** [comment]
This is looking really good @mpfeil. Thanks for helping out 🙌

I'm gonna merge this now as I need to do some conflicting work.

**Comment by sindresorhus** [comment]
@mpfeil My conflicting work is done. You can continue converting the tests if you want :)

File: test/helpers.js [hunk 3]
```diff
@@ -1,29 +1,16 @@
...
-import {createServer} from './helpers/server';
+import {withServer} from './helpers/with-server';
 
-let s;
-
-test.before('setup', async () => {
-	s = await createServer();
-
-	s.on('/', (request, response) => {
+test('promise mode', withServer, async (t, s) => {
```
Comment (sindresorhus) [inline test/helpers.js]: If you continue this, use `server` instead of `s`. It's more readable.

Comment (mpfeil) [reply test/helpers.js]: Going to change it in all tests.

**Pull request merged**